### PR TITLE
fix:Field switch should be locked even when the field already has values in RangePicker with showTime enabled

### PR DESCRIPTION
**Pull request by Zyf665**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx
 fix:https://github.com/ant-design/ant-design/issues/51693
### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.
problem:returns wrong dates onChange when manually change date time without pressing OK button
solution:Field switch should be locked even when the field already has the values

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Field switch should be locked even when the field already has values in RangePicker with showTime enabled   |
| 🇨🇳 Chinese |    修复带时间选择的日期范围选择器（showTime）在切换输入框时，即使输入框已有值也需要等待确认 |

## Summary by CodeRabbit

- **新功能**
	- 改进了日期范围选择器的值提交和验证逻辑，确保用户输入与日历值的一致性。
	- 引入了新的提交索引引用，以增强输入字段之间的焦点管理。
- **测试**
	- 添加了新的测试用例，以验证日期范围选择器在已有值时的焦点行为，确保用户体验的一致性。

### CODE REVIEW

File: src/PickerInput/hooks/useRangeValue.ts [hunk 1]
```diff
@@ -335,7 +334,10 @@ export default function useRangeValue<ValueType extends DateType[], DateType ext
 
   // ============================ Check =============================
   function hasSubmitValue(index: number) {
-    return !!submitValue()[index];
+    return (
+      !!submitValue()[index] &&
+      isSame(generateConfig, locale, submitValue()[index], getCalendarValue()[index], picker)
```
Comment (zombieJ) [inline src/PickerInput/hooks/useRangeValue.ts:339]: 这个不太靠谱，因为相同也可能是选择了就是相同的时间。应该是和那个 activeIndex 记录器一样，在设置时同步记录一下才对

File: src/PickerInput/hooks/useRangeValue.ts [hunk 2]
```diff
@@ -187,7 +186,7 @@ export default function useRangeValue<ValueType extends DateType[], DateType ext
   /** Trigger `onChange` directly without check `disabledDate` */
   triggerSubmitChange: (value: ValueType) => boolean,
   /** Tell `index` has filled value in it */
-  hasSubmitValue: (index: number) => boolean,
+  hasSubmitValue: (index: number, submitIndex: number | null) => boolean,
```
Comment (zombieJ) [inline src/PickerInput/hooks/useRangeValue.ts:189]: 耦合太高了，active 的 hooks 里可以加个方法来更新 submitValueRef 记录。并且自己提供一个 `hasActiveSubmitValue(index)`。这里不改，否则到处飞线。

Comment (Zyf665) [reply src/PickerInput/hooks/useRangeValue.ts]: 佬  hasSubmitValue这个方法  只有在这个文件里用了    但是在这一块没法在用这个方法了   我就在这个文件里把这个方法去掉了   现在的代码扫描过不了

File: src/PickerInput/RangePicker.tsx [hunk 3]
```diff
@@ -325,8 +327,6 @@ function RangePicker<DateType extends object = any>(
     flushSubmit,
     /** Trigger `onChange` directly without check `disabledDate` */
     triggerSubmitChange,
-    /** Tell `index` has filled value in it */
-    hasSubmitValue,
```
Comment (zombieJ) [inline src/PickerInput/RangePicker.tsx:329]: 这个既然没用了，是不是可以从 hooks 里删了？

Comment (Zyf665) [reply src/PickerInput/RangePicker.tsx:329]: ok

**Review by zombieJ — APPROVED** [review]

**Pull request merged**